astroquery/query.py

```diff
@@ -466,17 +472,26 @@ def _download_file(self, url, local_filepath, timeout=None, auth=None,
                 statinfo = os.stat(local_filepath)
                 if statinfo.st_size != length:
                     log.warning(f"Found cached file {local_filepath} with size {statinfo.st_size} "
-                                f"that is different from expected size {length}")
+                                f"that is different from expected size {length}.  ")
+                    if continuation:
+                        log.warning(
+                            "Continuation was requested but is not possible because "
+                            "'Accepts-Ranges' is not in the response headers.")
                     open_mode = 'wb'
+                    response = self._session.request(method, url,
+                                                     timeout=timeout, stream=True,
+                                                     auth=auth, **kwargs)
+                    response.raise_for_status()
                 else:
-                    log.info("Found cached file {0} with expected size {1}."
-                             .format(local_filepath, statinfo.st_size))
+                    log.info(f"Found cached file {local_filepath} with expected size {statinfo.st_size}.")
                     response.close()
-                    return
+                    return local_filepath
             else:
-                log.info("Found cached file {0}.".format(local_filepath))
+                # This case doesn't appear reachable under normal circumstances
+                # It is not covered by tests, and probably indicates a badly-behaved server
+                log.info(f"Found cached file {local_filepath}.  Could not verify length.")
```

Comment:
we may then want to raise an exception for these and even delete the cached file?

Reply:
that's not a bad idea, I'll do that.